File: infrastruktur/src/test/java/com/github/haschi/haushaltsbuch/testing/sandbox/_AussereKlasse.java
package com.github.haschi.haushaltsbuch.testing.sandbox;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.github.haschi.haushaltsbuch.infrastruktur.modellierung.de.Ereignis;

@Ereignis
@JsonDeserialize(as = AussereKlasse.class)
public interface _AussereKlasse
{
    KlasseA a();
    KlasseB b();
    KlasseC c();
    KlasseD d();
}

File: infrastruktur/src/test/kotlin/com/github/haschi/haushaltsbuch/testing/sandbox/_AussereKlasse.kt
package com.github.haschi.haushaltsbuch.testing.sandbox

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonUnwrapped
import com.fasterxml.jackson.databind.annotation.JsonDeserialize

@JsonDeserialize
data class AussereKlasse @JsonCreator constructor(
        val a: KlasseA,
        val b: KlasseB,
        val c: KlasseC,
        val d: KlasseD
)
